frame/election-provider-multi-phase/src/unsigned.rs

```diff
@@ -881,4 +929,75 @@ mod tests {
 			assert!(matches!(call, OuterCall::MultiPhase(Call::submit_unsigned(_, _))));
 		})
 	}
+
+	#[test]
+	fn trim_compact_length_does_not_modify_when_short_enough() {
+		let (mut ext, _) = ExtBuilder::default().build_offchainify(0);
+		ext.execute_with(|| {
+			roll_to(25);
+
+			let RoundSnapshot { voters, ..} =
+				MultiPhase::snapshot().unwrap();
+
+			let RawSolution { mut compact, .. } = raw_solution();
+			let encoded_len = compact.encode().len() as u32;
+			let compact_clone = compact.clone();
```

Comment:
This test is not complete, because the assumption upon which we assert is not enforced. For example, here, I see that calling `trim_compact_length` is a `noop`, and then from this I have to deduce that the length of the solution must have been less than the limit. This is the opposite of how it should be. 

So what you are missing here is a line like: 

```
assert!(compact.len() < T::MinerMaxLen)
```

A helpful pattern that I've seen from talented colleagues here is to split your tests (mentally or literally) into 3 stages: 
1. given (initial data and boilerplate)
2. when (prerequisites)
3. then (main assertion)

In this case, I'd expect to see

```
roll_to(25);

// given
let RoundSnapshot { voters, ..} = MultiPhase::snapshot().unwrap();
let RawSolution { mut compact, .. } = raw_solution();
let encoded_len = compact.encode().len() as u32;
let compact_clone = compact.clone();

// when
assert!(compact.len() <= T::MinerMaxLen)

// then 
compact = MultiPhase::trim_compact_length(
	encoded_len,
	compact,
	voter_index_fn_linear::<Runtime>(&voters),
).unwrap();
assert_eq!(compact, compact_clone);
```

Reply:
Ok, changed it as suggested in f86ae653b. 

However, I don't entirely agree that we depended on an untested assumption. Within the context of the test, we then trim it again, passing in the desired length explicitly: its current length. No matter how long or short the original `compact` is, `trim_compact_length` should not modify it under those circumstances.

Reply:
ahhhhh I see what you mean now, then `assert!(compact.len() <= T::MinerMaxLen)` is pretty pointless. 

Sorry, my mistake overall.